src/dune/exe_rules.ml

```diff
@@ -91,18 +91,51 @@ let executables_rules ~sctx ~dir ~expander ~dir_contents ~scope ~compile_info
      make the code more uniform. *)
   let ext_lib = ctx.lib_config.ext_lib in
   let link_args =
+    let paths_to_relative_foreign_archives () =
+      Command.of_result_map
+        (Lazy.force (Lib.Compile.requires_link compile_info))
+        ~f:(fun libs ->
+          Command.Args.S
+            (List.map libs ~f:(fun lib ->
+                 let lib_files = Lib_info.foreign_archives (Lib.info lib)
+                 and dll_files = Lib_info.foreign_dll_files (Lib.info lib) in
+                 Command.Args.S
+                   ( List.map lib_files ~f:(fun path ->
+                         match Path.parent path with
+                         (* TODO: Make the [None] case impossible by separating
+                            files and directories at the type level. Then every
+                            file will necessarily have the parent directory,
+                            possibly ".". All directories except for the root
+                            will have parents too. Same in the [dll_files] case. *)
+                         | None -> Command.Args.empty
+                         | Some dir ->
+                           (* TODO: Should be conditional on linkage. *)
+                           Command.Args.S
+                             [ A "-ccopt"
+                             ; A "-L"
+                             ; A "-ccopt"
+                             ; Path dir
+                             ; Hidden_deps (Dep.Set.singleton (Dep.file path))
+                             ])
+                   @ List.map dll_files ~f:(fun path ->
+                         match Path.parent path with
+                         | None -> Command.Args.empty
```

Comment:
I would suggest to use `Path.parent_exn` if this is indeed impossible. Same for the case above

Reply:
Good idea, done.